### DISCUSSION TITLE
Using flipper as permanent feature control

### DISCUSSION THREAD

**Discussion by dgmora** [jnunemaker/flipper#725] [Q&A] [answered]
Hi,

when reading through Flipper documentation, I see that it's mostly designed to be used as something temporal. But I was wondering if it's also used a a way to permanently control what users can access what features. Is it discouraged, or a well supported use case? I understand that in general it's not designed to be used with thousands of actors in mind.

My basic idea was that users belong to companies, and companies have a "plan". And a plan is composed by multiple features. Maybe using plans as actors would be a solution?

**Comment by jnunemaker** [comment, marked as answer]
> Is it discouraged, or a well supported use case?

Nope. I do it all the time. It's my favorite thing about flipper. I use it for permanent, long-lived features at least as much as temporary roll outs.

I wouldn't do thousands of actors per feature, but you are welcome to try. You'll likely want to turn off preloading (but leave memoization on). That'll ensure you are only loading all those actors for the features being checked. 

You are also in luck that we've just done several performance rounds to make loading lots of actors faster (active record adapter).

It's likely that what you are trying to do may be a little odd right now, but would be much easier once we get expressions #557  launched. We don't currently have a timeline for it but are hoping May/June.

**Reply by dgmora** [reply]
Nice! So how do you do that if there are many Users? Are groups then a company, or a 'plan' ?

**Reply by jnunemaker** [reply]
Yes, I'd use groups. Currently we do some things based on plan (e.g. basic, plus, premium). The code below is straight out of my config/initializers/flipper.rb file in one app. 

```ruby
Flipper.register(:admins) do |actor|
  actor.try(:admin?)
end

Flipper.register(:premium) do |actor|
  actor.try(:premium?)
end

Flipper.register(:plus) do |actor|
  actor.try(:plus?)
end

Flipper.register(:basic) do |actor|
  actor.try(:basic?)
end
```

Company groups would be annoying unless you have a very manual onboarding process and want to deploy each time a new company is added. 

If I knew more I could maybe suggest more but hopefully that is helpful.

**Reply by dgmora** [reply]
I understand actor here is a user,or company? Does that scale well with hundreds of them?
